src/churn-pipe/churn-pipe.ts

```diff
@@ -157,14 +135,21 @@ class Pipe {
         10);
   }
 
-  // Set the current transformer parameters.  The default is no transformation.
-  public setTransformer = (
-      transformerName :string,
-      key ?:ArrayBuffer,
-      config ?:string) :Promise<void> => {
+  // Set transformer parameters.
+  public setTransformer = (config:churn_types.ObfuscatorConfig) : Promise<void> => {
     try {
-      log.info('%1: using %2 transformer', this.name_, transformerName);
-      this.transformer_ = makeTransformer_(transformerName, key, config);
+      if (!(config.name in transformers)) {
+        throw new Error('unknown transformer: ' + config.name);
+      }
+
+      log.info('%1: using %2 obfuscator', this.name_, config.name);
+      this.transformer_ = new transformers[config.name]();
+      if (config.config !== undefined) {
+        this.transformer_.configure(config.config);
+      } else {
+        log.warn('%1: no config specified for obfuscator, using default', this.name_);
```

Comment:
Right. Let's say...this code is not exactly where I'd like it to be. An obvious improvement is to pass the `ObfuscatorConfig` to the constructor. I was punting on that but am happy to do it in this pull request.